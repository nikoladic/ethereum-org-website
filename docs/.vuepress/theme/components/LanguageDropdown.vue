<template>
  <DropdownLink :item="languages"/>
</template>

<script>
import DropdownLink from "./DropdownLink.vue";
import { translate } from "../utils/translations";

export default {
  components: { DropdownLink },
  computed: {
    languages() {
      const { locales } = this.$site;
      const currentLink = this.$page.path;
      const routes = this.$router.options.routes;
      const languageDropdown = {
        text: "Languages",
        items: Object.keys(locales).map(path => {
          const locale = locales[path];
          const text = locale.label
          let link;
          // Stay on the current page
          if (locale.lang === this.$lang) {
            link = currentLink;
          } else {
            // Try to stay on the same page
            link = currentLink.replace(this.$localeConfig.path, path);
            // fallback to homepage
            if (!routes.some(route => route.path === link)) {
              link = path;
            }
          }
          return { text, link };
        })
      };
      return languageDropdown;
    }
  }
};
</script>
