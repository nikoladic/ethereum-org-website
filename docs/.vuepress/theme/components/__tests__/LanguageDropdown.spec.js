import { shallowMount } from '@vue/test-utils'
import LanguageDropdown from "@/theme/components/LanguageDropdown"

describe('LanguageDropdown', () => {
  test('render with locales', () => {
    const wrapper = shallowMount(LanguageDropdown, {
      mocks: {
        $site: {
          locales: {
            "en": {
              label: "English",
              lang: "en"
            },
            "de": {
              label: "Deutsch",
              lang: "de"
            }
          },
        },
        $lang: "ko",
        $page: {
          path: "/en"
        },
        $localeConfig: {
          path: "en"
        },
        $router: {
          options: {
            routes: [{
              path: "/en"
            }, {
              path: "/de"
            }]
          }
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render without locales', () => {
    const wrapper = shallowMount(LanguageDropdown, {
      mocks: {
        $site: {
          locales: {
          },
        },
        $lang: "ko",
        $page: {
          path: "/en"
        },
        $localeConfig: {
          path: "en"
        },
        $router: {
          options: {
            routes: [{
              path: "/en"
            }, {
              path: "/de"
            }]
          }
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
