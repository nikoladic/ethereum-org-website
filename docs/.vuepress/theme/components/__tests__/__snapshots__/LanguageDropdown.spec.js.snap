// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageDropdown render with locales 1`] = `
<dropdownlink-stub
  item="[object Object]"
/>
`;

exports[`LanguageDropdown render without locales 1`] = `
<dropdownlink-stub
  item="[object Object]"
/>
`;
